clients/drcachesim/common/trace_entry.h

```diff
@@ -983,6 +987,120 @@ trace_arch_string(offline_file_type_t type)
                                                                       : "unspecified")));
 }
 
+/* Returns a string representation of marker type and corresponding marker value (if any)
+ * together.
+ */
+static inline std::string
+trace_marker_type_value_as_string(trace_marker_type_t marker_type, uintptr_t marker_value)
+{
+    std::stringstream ss;
+    const char *marker_name = trace_marker_names[marker_type];
+    switch (marker_type) {
+    /* Handle all the cases where marker_value doesn't matter.
+     */
+    case TRACE_MARKER_TYPE_FILTER_ENDPOINT:
+    case TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL:
+    case TRACE_MARKER_TYPE_CORE_WAIT:
+    case TRACE_MARKER_TYPE_CORE_IDLE: ss << "<" << marker_name << ">\n"; break;
+    /* Handle all the cases where we simply print <marker_type marker_value>.
+     */
+    case TRACE_MARKER_TYPE_TIMESTAMP:
+    case TRACE_MARKER_TYPE_CPU_ID:
+    case TRACE_MARKER_TYPE_INSTRUCTION_COUNT:
+    case TRACE_MARKER_TYPE_CACHE_LINE_SIZE:
+    case TRACE_MARKER_TYPE_PAGE_SIZE:
+    case TRACE_MARKER_TYPE_CHUNK_INSTR_COUNT:
+    case TRACE_MARKER_TYPE_SYSCALL:
+    case TRACE_MARKER_TYPE_DIRECT_THREAD_SWITCH:
+    case TRACE_MARKER_TYPE_WINDOW_ID:
+    case TRACE_MARKER_TYPE_SYSCALL_IDX:
+        ss << "<" << marker_name << " " << marker_value << ">\n";
+        break;
+    /* Handle all the cases where we simply print <marker_type 0xmarker_value>.
+     */
+    case TRACE_MARKER_TYPE_FUNC_RETADDR:
+    case TRACE_MARKER_TYPE_FUNC_ARG:
+    case TRACE_MARKER_TYPE_FUNC_RETVAL:
+    case TRACE_MARKER_TYPE_RECORD_ORDINAL:
+    case TRACE_MARKER_TYPE_SPLIT_VALUE:
+    case TRACE_MARKER_TYPE_BRANCH_TARGET:
+        ss << "<" << marker_name << " 0x" << std::hex << marker_value << std::dec
+           << ">\n";
+        break;
+    /* Handle all remaining cases where we want to print a more informative output.
+     */
+    case TRACE_MARKER_TYPE_SYSCALL_TRACE_START:
+    case TRACE_MARKER_TYPE_SYSCALL_TRACE_END:
+        ss << "<" << marker_name << " number " << marker_value << ">\n";
+        break;
+    case TRACE_MARKER_TYPE_CONTEXT_SWITCH_START:
+    case TRACE_MARKER_TYPE_CONTEXT_SWITCH_END:
+        ss << "<" << marker_name << " type " << marker_value << ">\n";
+        break;
+    /* We don't have a way to know the trace version here.  This might be an offset,
+     * but we don't make any distinction.
```

Comment:
Please elaborate as a new reader will have no idea what you mean: I only know b/c I remember a legacy version having offset instead of abs addr.